R/box_comment.R

```diff
@@ -0,0 +1,67 @@
+
+#' Create/delete Box comments
+#' 
+#' Comments can be associated with files or other comments. If a `comment_id` 
+#' is used it will be posted as a reply. Retreiving comments en bulk is only
+#' possible on a `file_id`. 
+#' 
+#' @md
+#' @inheritParams box_dl
+#' @param msg `character` the comment Note: tagging people with the
+#' \@user pattern is *not* supported at this time.
+#' @param comment_id `numeric` or `character`, comment ID at Box. 
+#' @return Invisible, the `comment_id` of the newly created Box comment.
+#' @examples 
+#' \dontrun{
+#' box_comment(12345, "Rmd report is ready for review")
+#' box_comment(comment_id = 98756, msg = "Responsed to your change requests!")
+#' }
+#' @export
+box_comment_create <- function(file_id = NULL, msg, comment_id = NULL) {
```

Comment:
one point that I should have picked-up earlier, would you consider naming the argument `comment`, vs. `msg`?

Reply:
I'm not set on `msg` but two things I considered: 
1) the api calls it "message"
2) down the road I would like to support another param `tag_users` that would insert the proper \@ tags as a prefix/suffix for notifying folks. This will require some accessory function to look people up by email-id, because box comments require the userID to be referenced. So I see the message text being combined with the tags to create a comment and wanted to have a clear distinction there. Currently inline \@-ing someone in the message seems like it is will be hard to pull off

Reply:
I think what you propose could be managed with a single parameter (maybe `message`?) 

For now `msg` has to be `character`.

In the future we could do something like:

```r
msg = box_tag("Looks good!", user_id = 12345)

msg = box_tag("Looks good!", box_search_user("Nathan Day"))
```

In either case, `box_tag()` could add an S3 class (maybe called `boxr_tag`) so that `box_comment_create()` would know to use the `tagged_message` parameter.

There would need to be an internal helper that would call the Box user-endpoint such that:

```r
box_create_tag <- function(user_id){
  # call user endpoint to get name

  glue::glue("[{user_id}:{name}]")
}
```

Reply:
I've renamed the param to `message`